File: Others/KonturIntegration/KonturWebExtension/src/ServerRequests.ts/KonturRequestController.ts
import { $RequestManager } from "@docsvision/webclient/System/$RequestManager";
import { serviceName } from "@docsvision/webclient/System/ServiceUtils";

export class KonturRequestController {
    
    constructor(private services: $RequestManager) {
    }

    getFromKontur(method: string, parameters: string): Promise<string | null> {
        return this.services.requestManager.get(`api/KonturRequest/GetFromKontur?method=${method}&parameters=${parameters}`);
    }

    addKonturReportToCard(cardId: string, reportFileName: string, parameters: string) {
        return this.services.requestManager.get(`api/KonturRequest/AddKonturReportToCard?cardId=${cardId}&fileName=${encodeURIComponent(reportFileName)}&parameters=${parameters}`);
    }
}

export type $KonturRequestController = {  konturRequestController: KonturRequestController }
export const $KonturRequestController = serviceName((x: $KonturRequestController) => x.konturRequestController);

File: Others/KonturIntegration/KonturServerExtension/Controllers/KonturRequestController.cs
using DocsVision.Platform.WebClient;
using Kontur.Services;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;

namespace Kontur.Controllers
{
    public class KonturRequestController : ControllerBase
    {
        readonly IKonturRequestService konturRequestService;
        readonly ICurrentObjectContextProvider currentObjectContextProvider;

        public KonturRequestController(IKonturRequestService konturRequestService, ICurrentObjectContextProvider currentObjectContextProvider)
        {
            this.konturRequestService = konturRequestService;
            this.currentObjectContextProvider = currentObjectContextProvider;
        }

        [HttpGet]
        public Task<string> GetFromKontur(string method, string parameters)
        {
            return this.konturRequestService.PreformGetFromContur(method, parameters);
        }

        [HttpGet]
        public Task AddKonturReportToCard(Guid cardId, string fileName, string parameters)
        {
            var sessionContext = currentObjectContextProvider.GetOrCreateCurrentSessionContext();
            return konturRequestService.AddKonturReportToCard(sessionContext, cardId, fileName, parameters);
        }
    }
}